### PR TITLE
Add checks to ensure top-level init is a function

### PR DESCRIPTION
**Pull request by sjamesr**
Fixes https://github.com/go-lang-plugin-org/go-lang-idea-plugin/issues/2195, fixes https://github.com/go-lang-plugin-org/go-lang-idea-plugin/issues/2190

### CODE REVIEW

**Comment by dlsniper** [comment]
@sjamesr This fixes https://github.com/go-lang-plugin-org/go-lang-idea-plugin/issues/2190 as well, no?

**Comment by sjamesr** [comment]
Thank you, amended the commit message.

File: src/com/goide/highlighting/GoAnnotator.java [hunk 1]
```diff
@@ -112,6 +112,53 @@ else if (element instanceof GoBuiltinCallExpr) {
         checkMakeCall(call, holder);
       }
     }
+    else if (element instanceof GoTopLevelDeclaration) {
+      if (element.getParent() instanceof GoFile) {
+        if (element instanceof GoTypeDeclaration) {
+          for (GoTypeSpec spec : ((GoTypeDeclaration)element).getTypeSpecList()) {
+            if ("init".equals(spec.getIdentifier().getText())) {
```
Comment (zolotov) [inline src/com/goide/highlighting/GoAnnotator.java]: spec.getIdentifier().textMatches(GoConstants.INIT)

File: src/com/goide/highlighting/GoAnnotator.java [hunk 2]
```diff
@@ -112,6 +112,53 @@ else if (element instanceof GoBuiltinCallExpr) {
...
+            }
+          }
+        }
+        else if (element instanceof GoFunctionOrMethodDeclaration) {
+          if (element instanceof GoFunctionDeclaration) {
+            GoFunctionDeclaration declaration = (GoFunctionDeclaration)element;
+            if ("init".equals(declaration.getIdentifier().getText())) {
+              GoSignature signature = declaration.getSignature();
+              if (signature != null) {
+                if (declaration.getSignature().getResult() != null) {
```
Comment (zolotov) [inline src/com/goide/highlighting/GoAnnotator.java]: you're already have signature variable here

File: src/com/goide/highlighting/GoAnnotator.java (hunk 1 above)
Comment (sjamesr) [inline src/com/goide/highlighting/GoAnnotator.java]: Fixed here and elsewhere, thanks

File: src/com/goide/highlighting/GoAnnotator.java (hunk 2 above)
Comment (sjamesr) [inline src/com/goide/highlighting/GoAnnotator.java]: Thank you, fixed

**Comment by sjamesr** [comment]
Rebased

**Comment by zolotov** [comment]
Merged

**Pull request closed without merging**